### DISCUSSION TITLE
`reportRedeclaration` diagnostics for redeclared property setters and deleters

### DISCUSSION THREAD

**Discussion by InSyncWithFoo** [microsoft/pyright#8166] [General]
Reproducible example ([playground](https://pyright-play.net/?pyrightVersion=1.1.367&pythonVersion=3.13&strict=true&code=MYGwhgzhAEDCBcBYAUCgkGe0CWA7ALtEcSaWeQMTQAiApqGAE5j7YD2u0ARGFzjGwBGEYAFdGtACbRBAT2hhok%2BuGasO0NgDNo%2BABa1oEMAFtDuU7XSYj%2BRinSCseQuTfviVAHJtotRoxsjBCOzgQOyGjKOiYAFBC0IFoAlNAAtAB8toxYAHT5HtBUALK0%2BmzSygxq7JxcJnzYAsJiEtJyCkoqTCy1mjr6hsZm0BZm6NHQcQlJqZnZefkRaAACAA6Ba-74shO0OmvxiSnpWS6LuYUlZXoVXdW9Glxrjc0i4lIy8opVqo%2Bc2l0BiMllGlnQ602212kUmhxmJ3m52g%2BVyy3WuQS%2BHw-j2ByOSQANNAAG5hfBzLI%2BXC0C5Eby%2BfyBYIQtaYso4%2Byw-bQeHHYlknAESnQam0lFLVCRDHKEBlXHc-EIgXkkViulFUWMgJBELStmy%2BVcqI8vlE0mq05amkXFBAA)):

```python
class C:

	def m(self) -> str: ...           # Method declaration "m" is obscured by a declaration of the same name
	def m(self) -> str: ...

	@property
	def p(self) -> int: ...           # Method declaration "p" is obscured by a declaration of the same name
	@property
	def p(self) -> int: ...
```

```python
	@p.setter
	def p(self, v: int) -> None: ...  # No errors
	@p.setter
	def p(self, v: str) -> None: ...

	@p.deleter
	def p(self, v: int) -> None: ...  # No errors
	@p.deleter
	def p(self, v: str) -> None: ...
```

Since at runtime `@p.setter` just overwrites `C.p` with a new `property` object containing the latest setter method, I think it would make sense to consider this as a redeclaration and report it accordingly.

Instance attribute redeclarations with the same annotations also don't trigger this rule, but this seems to be intentional.

```python
	a: int                            # Declaration "a" is obscured by a declaration of the same name
	a: str

	b: int                            # No errors
	b: int
```

**Comment by erictraut** [comment]
Pyright is working as intended here. While you're correct that `@p.setter` overwrites `C.p` with a new property object, this is such a common pattern that pyright (and other type checkers like mypy) special case it and do not treat it like a redeclaration.
